### PR TITLE
[macOS] Pre-cache Ruby binaries

### PR DESCRIPTION
**Pull request by dibir-magomedsaygitov**
# Description
In scope of this PR we are pre-caching Ruby binaries on our macOS images. See details in related issue.

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by vsafonkin — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

File: images/macos/provision/core/ruby.sh [hunk 1]
```diff
@@ -4,3 +4,36 @@ source ~/utils/utils.sh
 
 echo Installing Ruby...
 brew install ruby
+
+echo "Install Ruby..."
+PACKAGE_TAR_NAMES=$(curl -s "https://api.github.com/repos/ruby/ruby-builder/releases/latest" | jq -r '.assets[].name')
+TOOLSET_VERSIONS=$(get_toolset_value '.toolcache[] | select(.name | contains("Ruby")) | .versions[]')
+TOOLSET_PLATFORM=$(get_toolset_value '.toolcache[] | select(.name | contains("Ruby")) | .platform')
```
Comment (maxim-lobanov) [inline images/macos/provision/core/ruby.sh:11]: Do we need to read it from toolset? I guess `platform` is identical for all macOS versions.

File: images/macos/toolsets/toolcache-10.13.json [hunk 2]
```diff
@@ -1,8 +1,5 @@
 {
     "@actions/toolcache-python-macos-1013-x64": [
```
Comment (maxim-lobanov) [inline images/macos/toolsets/toolcache-10.13.json:2]: On High Sierra, this file is not used anymore. Please remove it fully

File: images/macos/provision/core/ruby.sh (hunk 1 above)
Comment (dibir-magomedsaygitov) [reply images/macos/provision/core/ruby.sh]: Thank you. Not really. I removed `platform`  from toolset.

File: images/macos/toolsets/toolcache-10.13.json (hunk 2 above)
Comment (dibir-magomedsaygitov) [reply images/macos/toolsets/toolcache-10.13.json:2]: Thank you, file is fully removed.

**Review by MaksimZhukov — APPROVED** [review]

**Review by AlenaSviridenko — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Pull request merged**